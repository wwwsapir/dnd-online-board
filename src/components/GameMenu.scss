@import ".common";

.menu-window .GameMenu-window {
  left: 22%;
  right: 22%;
  top: 20%;
  bottom: 20%;
}
