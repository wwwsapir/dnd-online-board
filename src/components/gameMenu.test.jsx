import GameMenu from "./gameMenu";

it("renders correctly with authToken", () => {
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={() => {}}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("renders correctly without authToken", () => {
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={() => {}}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={""}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

jest.mock("../services/getUserGameData");
it("game data check functionality (async)", async (done) => {
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={() => {}}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );
  setTimeout(() => {
    wrapper.update();
    expect(wrapper.find("button").first().prop("disabled")).toEqual(false);
    done();
  });
});

it("start a new game click and warning menu opens (async)", (done) => {
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={() => {}}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );

  setTimeout(() => {
    wrapper.update();
    wrapper.find("button").at(1).simulate("click");
    expect(wrapper).toMatchSnapshot();
    done();
  });
});

it("start a new game click and user approval functionality (async)", (done) => {
  const spyOnStartANewGame = sinon.spy();
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={spyOnStartANewGame}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );

  setTimeout(() => {
    wrapper.update();
    wrapper.find("button").at(1).simulate("click");
    wrapper.find("button").at(2).simulate("click");
    expect(spyOnStartANewGame.calledOnce).toBe(true);
    done();
  });
});

it("start a new game click and user rejection functionality (async)", (done) => {
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={() => {}}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );

  setTimeout(() => {
    wrapper.update();
    wrapper.find("button").at(1).simulate("click");
    wrapper.find("button").at(3).simulate("click");
    expect(wrapper).toMatchSnapshot();
    done();
  });
});

it("cancel redirect from map at mount functionality", () => {
  const spyCancelRedirectFromMap = sinon.spy();
  shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={() => {}}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={spyCancelRedirectFromMap}
      isSmallerScreen={false}
    />
  );

  expect(spyCancelRedirectFromMap.calledOnce).toBe(true);
});

it("log out link click functionality", () => {
  const spyOnLogOut = sinon.spy();
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={() => {}}
      onStartANewGame={() => {}}
      onLogOut={spyOnLogOut}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );

  wrapper.find("a").simulate("click");
  expect(spyOnLogOut.calledOnce).toBe(true);
});

it("continue game click functionality (async)", (done) => {
  const spyOnContinueSavedGame = sinon.spy();
  const wrapper = shallow(
    <GameMenu
      onContinueSavedGame={spyOnContinueSavedGame}
      onStartANewGame={() => {}}
      onLogOut={() => {}}
      userName={"userName"}
      authToken={"authToken"}
      cancelRedirectFromMap={() => {}}
      isSmallerScreen={false}
    />
  );

  setTimeout(() => {
    wrapper.update();
    wrapper.find("button").first().simulate("click");
    expect(spyOnContinueSavedGame.calledOnce).toBe(true);
    done();
  });
});
