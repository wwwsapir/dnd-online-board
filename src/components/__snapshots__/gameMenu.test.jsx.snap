// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with authToken 1`] = `
<div
  className="GameMenu-window"
>
  <ul
    className="w-100 menu"
  >
    <h4
      className="mb-4"
    >
      <span
        className="menu-header"
      >
        Hi, 
        userName
        ! Ready to play?
      </span>
    </h4>
    <li>
      <button
        className="btn btn-primary form-control mt-3"
        disabled={true}
        onClick={[Function]}
      >
        Collecting game data from server...
      </button>
    </li>
    <li>
      <button
        className="btn btn-primary form-control mt-3"
        disabled={true}
        onClick={[Function]}
      >
        Start a New Game
      </button>
    </li>
    <li
      className="mt-2 mb-4"
    >
      <a
        className="float-right"
        href="#"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders correctly without authToken 1`] = `
<div
  className="GameMenu-window"
>
  <ul
    className="w-100 menu"
  >
    <h4
      className="mb-4"
    >
      <span
        className="menu-header"
      >
        Hey... unknown user!
      </span>
      <br />
      <br />
      Oops! You can't see the game menu if you're not logged in :) 
      <br />
      Please log in to play.
    </h4>
    <Link
      to="/home/login"
    >
      Login Here
    </Link>
  </ul>
</div>
`;

exports[`start a new game click and user rejection functionality (async) 1`] = `
<div
  className="GameMenu-window"
>
  <ul
    className="w-100 menu"
  >
    <h4
      className="mb-4"
    >
      <span
        className="menu-header"
      >
        Hi, 
        userName
        ! Ready to play?
      </span>
    </h4>
    <li>
      <button
        className="btn btn-primary form-control mt-3"
        disabled={false}
        onClick={[Function]}
      >
        Continue Last Saved Game
      </button>
    </li>
    <li>
      <button
        className="btn btn-primary form-control mt-3"
        disabled={false}
        onClick={[Function]}
      >
        Start a New Game
      </button>
    </li>
    <li
      className="mt-2 mb-4"
    >
      <a
        className="float-right"
        href="#"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;

exports[`start a new game click and warning menu opens (async) 1`] = `
<div
  className="GameMenu-window"
>
  <ul
    className="w-100 menu"
  >
    <h4
      className="mb-4"
    >
      <span
        className="menu-header"
      >
        Hi, 
        userName
        ! Ready to play?
      </span>
    </h4>
    <li>
      <button
        className="btn btn-primary form-control mt-3"
        disabled={false}
        onClick={[Function]}
      >
        Continue Last Saved Game
      </button>
    </li>
    <li>
      <button
        className="btn btn-primary form-control mt-3"
        disabled={false}
        onClick={[Function]}
      >
        Start a New Game
      </button>
    </li>
    <li
      className="mt-2 mb-4"
    >
      <a
        className="float-right"
        href="#"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
    <span>
      <li
        className="mt-2"
      >
        <h6>
          Starting a new game will erase the last saved game, continue?
        </h6>
      </li>
      <span
        className="inline-form row"
      >
        <dev
          className="col-7 d-inline p-1"
        >
          <button
            className="btn btn-danger form-control mt-3"
            onClick={[Function]}
          >
            Yes - Delete my saved game
          </button>
        </dev>
        <dev
          className="col-5 d-inline p-1"
        >
          <button
            className="btn btn-primary form-control mt-3"
            onClick={[Function]}
          >
            No - Keep my saved game
          </button>
        </dev>
      </span>
    </span>
  </ul>
</div>
`;
